doc/configuration.rst

```diff
@@ -588,6 +589,20 @@ a default::
 The highest precedence is always given to the `-D` flag of the
 ``sphinx-build`` command.
 
+.. _rebuild_changed_only:
+
+Only rebuild newly changed examples
+===================================
+
+By default, Sphinx-Gallery will always generate rst files and backreferences
+for all examples. This can make the re-building process slow for projects with
+a lot of examples. If you want to only generate rst files for newly updated
+examples, you can set the ``rebuild`` parameter to ``'mtime'``::
+
+    sphinx_gallery_conf = {
+        ...
+        'rebuild': 'mtime',
+    }
```

Comment:
Can you also add a `.. note::` at the bottom of the `Parsing and executing examples via matching patterns` section where it talks about building a specific example that if people's `conf.py` has `'rebuild': 'mtime'`, they might also need to set `-D sphinx_gallery_conf.rebuild=always`?

Also maybe a complementary note in this section would help. I know there are some repos that use this `-D` pattern to build specific files in a `Makefile`, and it would be good to give them a hint that if what they want is to build e.g. one file no matter what (say they are modifying the library, not the example) then they'll want to add another `-D` to override `rebuild`, too.